compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs

```diff
@@ -938,11 +938,6 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     ) {
                         // Include previous comma
                         span = span.with_lo(prev.hi());
```

Comment:
```suggestion
                        span = prev.shrink_to_hi().to(span);
```

Otherwise this will cause [cursed spans for code with macros](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021).

Reply:
Applying the fix - but btw the playground link is missing a gist

Reply:
Whoops, fixed link.

Here's the code and the current output:
```rust
macro_rules! foo {
    ($x:expr) => {
        bar($x, 1);
    }
}

fn bar(_: i32) {}

fn main() {
    foo!(1);
}
```

```text
Compiling playground v0.0.1 (/playground)
error[[E0061]](https://doc.rust-lang.org/nightly/error_codes/E0061.html): this function takes 1 argument but 2 arguments were supplied
  --> src/main.rs:3:9
   |
3  |           bar($x, 1);
   |  _________^^^_____--
   | |                 |
   | |                 unexpected argument of type `{integer}`
4  | |     }
5  | | }
6  | |
...  |
9  | | fn main() {
10 | |     foo!(1);
   | |     -------
   | |_____|____|
   |       |    help: remove the extra argument
   |       in this macro invocation
   |
note: function defined here
  --> src/main.rs:7:4
   |
7  | fn bar(_: i32) {}
   |    ^^^ ------
   = note: this error originates in the macro `foo` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0061`.
error: could not compile `playground` due to previous error
```

Reply:
Added a test for the macro case too just in case